src/app/lib/features/vaccineRegistration/vaccineRegistrationSlice.tsx

```diff
@@ -0,0 +1,44 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from '../../store';
+import { VaccineRegistrationFormDataType } from '@/app/(mainPage)/portal/vaccine-registration/page';
+
+const initialState: PersonalInfoType = {
+  priorityOptions: '',
+  job: '',
+  workplace: '',
+  address: '',
+  desireddate: '',
+  healthInsuranceNumber: '',
+  appointmentDate: ''
+};
+
+export type PersonalInfoType = {
+  priorityOptions: string;
+  job: string;
+  workplace: string;
+  address: string;
+
+  appointmentDate: string;
+  desireddate: string;
+  healthInsuranceNumber: string;
+};
+
+export const vaccineRegistrationSlice = createSlice({
+  name: 'vaccineRegistration',
+  initialState,
+  reducers: {
+    submitFormData: (state, action) => {
+      const data = action.payload as PersonalInfoType;
+      return { ...state, ...data };
```

Comment:
redux toolkit hỗ trợ mutate state trực tiếp nhé https://redux-toolkit.js.org/usage/immer-reducers#mutating-and-returning-state